jep/0000/README.adoc

```diff
@@ -0,0 +1,348 @@
+= JEP-0000: WebSocket Services
+:toc: preamble
+:toclevels: 3
+ifdef::env-github[]
+:tip-caption: :bulb:
+:note-caption: :information_source:
+:important-caption: :heavy_exclamation_mark:
+:caution-caption: :fire:
+:warning-caption: :warning:
+endif::[]
+
+.Metadata
+[cols="1h,1"]
+|===
+| JEP
+| 0000
+
+| Title
+| WebSocket Services
+
+| Sponsor
+| link:https://github.com/jglick[Jesse Glick]
+
+// Use the script `set-jep-status <jep-number> <status>` to update the status.
+| Status
+| Not Submitted :information_source:
+
+| Type
+| Standards
+
+| Created
+| 2019-11-21
+
+| BDFL-Delegate
+| TBD
+
+|===
+
+== Abstract
+
+Jenkins offers an API allowing WebSocket services to be implemented.
+This is available using the Stapler `HttpResponse` idiom,
+served as the final action after the regular Stapler routing mechanism.
+Unlike regular HTTP endpoints, bidirectional streaming communication is possible.
+
+The principal service exposed in this way is a new Remoting transport for inbound build agents at an HTTP(S) URL.
+The agent itself (typically named `slave.jar`) includes a WebSocket client capable of connecting to this endpoint.
```

Comment:
I think the Jenkins project is trying to move people to use `agent.jar` instead of `slave.jar`.

Reply:
Yes, the current launch script in the Docker image reflects the rename. I will update this accordingly.